executor-service/src/main/java/executor/service/service/proxy/ProxySourcesClientService.java

```diff
@@ -52,7 +56,7 @@ public ProxyConfigHolder getProxy() {
         proxyConfigHolder.setProxyCredentials(credentialsQueue.poll());
```

Comment:
The `getProxy` is not thread-safe. If multiple threads call at the same time, they could get inconsistent results or even cause a `ConcurrentModificationException`